geopandas/array.py

```diff
@@ -228,10 +229,12 @@ class GeometryArray(ExtensionArray):
 
     _dtype = GeometryDtype()
 
-    def __init__(self, data, crs=None):
+    def __init__(self, data, crs=None, sindex_backend=None):
```

Comment:
Currently, we do not have `sindex_backend` option and pick the backend automatically, right? Can we do the same in array for now? This already expects flexible spatial index API, but there has been no decision on its design yet. Or is this needed? This PR should _just_ move the existing spatial indexing to array level, I don't thin we should expose multiple backends yet.

Reply:
Agreed, this was left over. I'll remove the constructor parameter, but leave the `self.sindex_backend` attribute to avoid future refactoring (the alternative is to call `get_sindex_class()` each time from within `GeometryArray.sindex`.)

Reply:
We do not know how the final API will look like. I would rather not assume the future now, since we do not know how it will look like and when it lands. So I'd prefer calling `get_sindex_class()` as we do in GeoDataFrame now.

Reply:
Understood. I implemented calling `get_sindex_class()` each time.